### PR TITLE
Fix not possible to resolve conflict in code editor

### LINKED ISSUE #1940: (Exam Mode) Not possible to resolve conflict after opening online editor and cloning repository
### Describe the bug
In test exam mode, when (accidentally) opening the online editor and then cloning the repository and pushing your commits, the repo enters a "conflict state". It is not possible to resolve this conflict, because clicking on "resolve conflict" shows a red "not found" message.

#### To Reproduce
1. Start test exam mode
2. Go to programming exercise
3. Open a file in the online editor
4. Clone the repository
5. Make changes and push the repository
6. Go to programming exercise
7. Click on "resolve conflict"
8. See error

#### Expected behavior
The conflict is resolved and it is possible to ignore the changes made in the online editor.

#### Screenshots
Screenshot 2020-07-20 at 08 53 35
Screenshot 2020-07-20 at 08 53 58

### Environment

<details><pre>
 - OS: MacOS 10.15.5
 - Browser Google Chrome (84.0.4147.89)
</pre></details>

### PR DESCRIPTION
**Pull request by schultek**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.

### Motivation and Context

Resolves #1940 

### Description

The Resolve Conflict Button wasn't working for exam exercises.

### Steps for Testing

1. Create an exam with programming exercises
2. Participate and edit a file in the online editor, save but don't commit
4. Clone the repository
5. Make local changes to the same file and push the repository (provoking a merge conflict)
5. Go to the online editor (you may have to click 'Refresh')
6. Click on "Resolve conflict"

### CODE REVIEW

**Comment by krusche** [comment]
What was the exact issue? Why did it state "Not found"?

**Comment by schultek** [comment]
There was an issue with the domain dependent service, it did not generate the correct url. I did not find the exact cause why this did not initialize correctly, so i wrote the resolve dialog similar to how the refresh button works. The reset function of the repository service is now called from the actions component and not from inside the modal, which somehow works.

**Comment by kloessst** [comment]
```
@Component({
    selector: 'jhi-code-editor-resolve-conflict-modal',
    templateUrl: './code-editor-resolve-conflict-modal.component.html',
    styleUrls: ['./code-editor-resolve-conflict-modal.scss'],
    providers: [CodeEditorRepositoryFileService],
})
export class CodeEditorResolveConflictModalComponent {
```
This is the root cause. We provide another instance of CodeEditorRepositoryFileService which won't be properly set up, as we provide all domain dependent services in ExamCodeEditorStudentContainerComponent and the domain is set for those services. The refactoring of @schultek moves the service call to a component which uses the properly configured service instance.

File: src/main/webapp/app/exercises/programming/shared/code-editor/actions/code-editor-resolve-conflict-modal.component.ts [hunk 1]
```diff
@@ -13,16 +13,16 @@ import { GitConflictState } from 'app/exercises/programming/shared/code-editor/m
 export class CodeEditorResolveConflictModalComponent {
     constructor(public activeModal: NgbActiveModal, private repositoryService: CodeEditorRepositoryService, private conflictService: CodeEditorConflictStateService) {}
```
Comment (kloessst) [inline src/main/webapp/app/exercises/programming/shared/code-editor/actions/code-editor-resolve-conflict-modal.component.ts:14]: CodeEditorRepositoryService and CodeEditorConflictStateService can be removed, the provided service as well

**Pull request merged**